tests/raftstore/test_service.rs

```diff
@@ -0,0 +1,367 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use grpc::{ChannelBuilder, Environment};
+use tikv::util::HandyRwLock;
+
+use kvproto::tikvpb_grpc::TikvClient;
+use kvproto::kvrpcpb::*;
+use kvproto::raft_serverpb::*;
+use futures::{Future, Sink};
+use kvproto::coprocessor::*;
+
+use super::server::*;
+
+#[test]
+fn test_grpc_service() {
+    let count = 1;
+    let mut cluster = new_server_cluster(0, count);
+    cluster.run();
+
+    let region_id = 1;
+    let leader = cluster.leader_of_region(region_id).unwrap();
+    let epoch = cluster.get_region_epoch(region_id);
+    let mut ctx = Context::new();
+    ctx.set_region_id(region_id);
+    ctx.set_peer(leader.clone());
+    ctx.set_region_epoch(epoch);
+
+    let addr = cluster.sim.rl().get_addr(leader.get_store_id());
+    let env = Arc::new(Environment::new(1));
+    let channel = ChannelBuilder::new(env).connect(&format!("{}", addr));
+    let client = TikvClient::new(channel);
+    let (k, v) = (b"key".to_vec(), b"value".to_vec());
+
+    // Raw KV
```

Comment:
Can you split it into several small test cases?

Reply:
+1